debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/RomClassWalker.java

```diff
@@ -846,6 +860,76 @@ int allSlotsInVerificationTypeInfoDo(U8Pointer cursor) throws CorruptDataExcepti
 		}
 		return 3;
 	}
+
+	void recordAttributeDo(U32Pointer attribute) throws CorruptDataException
+	{
+		if (attribute.isNull()) {
+			return;
+		}
+		U32Pointer attributeStart = attribute;
+		int numRecordComponents = attribute.at(0).intValue();
+		classWalkerCallback.addSlot(clazz, SlotType.J9_U32, attribute, "numberRecordComponents");
+		attribute = attribute.add(1);
+
+		for (int i = 0; i < numRecordComponents; i++) {
+			J9ROMRecordComponentShapePointer recordComponent = J9ROMRecordComponentShapePointer.cast(attribute);
+			attribute = U32Pointer.cast(recordComponent.add(1));
+
+			/* name and descriptor */
+			classWalkerCallback.addSlot(clazz, SlotType.J9_SRPNAS, recordComponent.nameAndSignature(), "nameAndSignature");
+			classWalkerCallback.addSlot(clazz, SlotType.J9_U32, recordComponent.attributeFlagsEA(), "attributeFlags");
+			classWalkerCallback.addSection(clazz, recordComponent, attribute.getAddress() - recordComponent.getAddress(), "recordComponentShape", true);
+
+			/* process variable attributes */
+			UDATA attributeFlags = recordComponent.attributeFlags();
+
+			if (attributeFlags.anyBitsIn(J9RecordComponentFlagHasGenericSignature)) {
+				classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, attribute, "recordComponentGenSigUTF8");
+				attribute = attribute.add(1);
+			}
+
+			if (attributeFlags.anyBitsIn(J9RecordComponentFlagHasAnnotations)) {
+				int increment = allSlotsInAnnotationDo(attribute, "recordComponentAnnotation");
+				attribute = attribute.add(increment);
+			}
+
+			if (attributeFlags.anyBitsIn(J9RecordComponentFlagHasTypeAnnotations)) {
+				int increment = allSlotsInAnnotationDo(attribute, "recordComponentTypeAnnotation");
+				attribute = attribute.add(increment);
+			}
+		}
+		
```

Comment:
Please remove the trailing whitespace.

Reply:
It's still here.

Reply:
Thanks. It is removed now.

Reply:
I see you've completely removed the lines containing only whitespace: it's not what I was asking for, but that's ok.